zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -104,32 +113,22 @@ public static Either<ProblemDetail, SearchQueryPage> toSearchQueryPage(
     return Either.right(null);
   }
 
-  public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
-      final List<SearchQuerySortRequest> sorting) {
+  private static <T, B extends AbstractBuilder<B> & ObjectBuilder<T>>
+      Either<ProblemDetail, T> toSearchQuerySort(
+          final List<SearchQuerySortRequest> sorting,
+          final Supplier<B> builderSupplier,
+          BiConsumer<String, B> sortFieldConsumer) {
     if (sorting != null && !sorting.isEmpty()) {
-      final var builder = SortOptionBuilders.processInstance();
-
+      final var builder = builderSupplier.get();
       for (SearchQuerySortRequest sort : sorting) {
-        final var field = sort.getField();
-        final var order = sort.getOrder();
-
-        if ("processInstanceKey".equals(field)) {
-          builder.processInstanceKey();
-        } else if ("startDate".equals(field)) {
-          builder.startDate();
-        } else if ("endDate".equals(field)) {
-          builder.endDate();
-        } else {
-          throw new RuntimeException("unkown sortBy " + field);
+        if (sort.getField() == null) {
+          throw new IllegalArgumentException("Sort field must not be null");
```

Comment:
❌ We should collect validation errors like this instead of throwing an exception here. The collected errors can then be returned to the user. @nathansandi worked through this over in this PR already, maybe the two of you can align here. Let's incorporate this here so @nathansandi can consume this in his PR then.

Reply:
Having test cases will help us validate this easily. I think you can mark fields as required in OpenAPI, generate the model, and have Spring throw an exception if the input is missing. 

It might be simpler to just incorporate validation errors here correctly in this PR, so @nathansandi's PR can be simplified a lot (just reusing your method for user task search accordingly).

Reply:
validationErrors are now collected